### PR TITLE
Move account fetching to run concurrently

### PR DESCRIPTION
**Pull request by svozza**
Description of changes:

When operating in `AWS_ORGANIZATION` mode, accounts are retrieved using the `ListAccount` API. The accounts are then persisted in the Settings DynamoDB table. However, the `ListAccount` API has a maximum page size of 20, meaning that for organizations with a large number of accounts gathering the list of the accounts could take some time. 

Compounding this issue is that account retrieval was done in the `initialisation` module, which meant that no processing could begin until it was finished. This PR moves the account retrieval logic out of the initialisation phase so it can run concurrently with the code that loads resources and relationships from the database and also queries the AWS Config aggregator.

The account retrieval code now also does a diff between what's returned from `ListAccounts` and what's stored in the DDB, any accounts in the database but not returned by `ListAccount` are deemed to have been deleted thus fixing #396 .

### CODE REVIEW

File: source/backend/discovery/src/lib/apiClient/index.js [hunk 1]
```diff
@@ -79,25 +86,124 @@ function updateCrawledAccounts(appSync) {
...
+        appSyncClient.getAccounts(),
+        organizationsClient.getAllAccounts(),
+        configClient.getConfigAggregator(configAggregator),
+        ec2Client.getAllRegions()
+    ]);
+
+    const dbAccountsMap = new Map(dbAccounts.map(x => [x.accountId, x]));
+    const orgAccountsMap = new Map(orgAccounts.map(x => [x.Id, x]));
+
+    const deletedAccounts = dbAccounts.reduce((acc, account) => {
```
Comment (tmekari) [inline source/backend/discovery/src/lib/apiClient/index.js:118]: Are deleted accounts accounts that were in the db before but are now gone when checking again?

File: source/backend/discovery/src/lib/apiClient/index.js [hunk 2]
```diff
@@ -79,25 +86,124 @@ function updateCrawledAccounts(appSync) {
...
+        }
+        return acc;
+    }, []);
+
+    const managementAccountId = parseArn(managementAccountArn).accountId;
+
+    return orgAccounts
+        .map(({Id, Name: name, Arn}) => {
+            const [, organizationId] = parseArn(Arn).resource.split('/');
+            const lastCrawled = dbAccountsMap.get(Id)?.lastCrawled;
```
Comment (tmekari) [inline source/backend/discovery/src/lib/apiClient/index.js:131]: What does last crawled contain?

**Review by tmekari — APPROVED** [review]

File: source/backend/discovery/src/lib/apiClient/index.js (hunk 1 above)
Comment (svozza) [reply source/backend/discovery/src/lib/apiClient/index.js:118]: Yeah, they're accounts in the DB but that aren't in the response from the `getAllAccounts` function that calls the AWS Organizations `listAccounts` API.

File: source/backend/discovery/src/lib/apiClient/index.js (hunk 2 above)
Comment (svozza) [reply source/backend/discovery/src/lib/apiClient/index.js:131]: The timestamp of the last time the discovery process successfully discovered the account.

**Pull request merged**